numcodecs/sz.py

```diff
@@ -0,0 +1,84 @@
+import ctypes
+from numcodecs.abc import Codec
+
+
+class Params(ctypes.Structure):
+
+    _fields_ = [
+        ("options_mask", ctypes.c_int),
+        ("bits_per_pixel", ctypes.c_int),
+        ("pixels_per_block", ctypes.c_int),
+        ("pixels_per_scanline", ctypes.c_int)
+    ]
+
+
+libsz = False
+for ext in [".so", ".dylib", ".dll"]:
+    try:
+        libsz = ctypes.cdll.LoadLibrary("libsz" + ext)
+    except OSError:
+        pass
+
+if libsz:
+    BufftoBuffCompress = libsz.SZ_BufftoBuffCompress
+    BufftoBuffCompress.restype = ctypes.c_int
+    BufftoBuffCompress.argtypes = [
+        ctypes.c_void_p, ctypes.POINTER(ctypes.c_size_t), ctypes.c_void_p, ctypes.c_size_t,
+        ctypes.POINTER(Params)
+    ]
+
+    BufftoBuffDecompress = libsz.SZ_BufftoBuffDecompress
+    BufftoBuffCompress = libsz.SZ_BufftoBuffCompress
+    BufftoBuffCompress.restype = ctypes.c_int
+    BufftoBuffCompress.argtypes = [
+        ctypes.c_void_p, ctypes.POINTER(ctypes.c_size_t), ctypes.c_void_p, ctypes.c_size_t,
+        ctypes.POINTER(Params)
+    ]
```

Comment:
```suggestion
    BufftoBuffDecompress = libsz.SZ_BufftoBuffDecompress
    BufftoBuffDecompress.restype = ctypes.c_int
    BufftoBuffDecompress.argtypes = [
        ctypes.c_void_p, ctypes.POINTER(ctypes.c_size_t), ctypes.c_void_p, ctypes.c_size_t,
        ctypes.POINTER(Params)
    ]
```

Reply:
Hah, that was silly.
Still crashed, though, even after my fixed type